eureka-client/src/main/java/com/netflix/discovery/shared/EurekaHttpClient.java

```diff
@@ -0,0 +1,57 @@
+package com.netflix.discovery.shared;
+
+import com.netflix.appinfo.InstanceInfo;
+import com.netflix.appinfo.InstanceInfo.InstanceStatus;
+
+/**
+ * Low level Eureka HTTP client API.
+ *
+ * @author Tomasz Bak
+ */
+public interface EurekaHttpClient {
+
+    HttpResponse<Void> register(InstanceInfo info);
+
+    HttpResponse<Void> cancel(String appName, String id);
+
+    HttpResponse<InstanceInfo> sendHeartBeat(String appName, String id, InstanceInfo info, InstanceStatus overriddenStatus);
```

Comment:
the InstanceInfo should contain the appName and id, so no need to pass these as params.